packages/edit-site/src/index.js

```diff
@@ -33,12 +34,37 @@ import List from './components/list';
  * @param {?Object} settings Editor settings object.
  */
 export function reinitializeEditor( target, settings ) {
+	// This will be a no-op if the target doesn't have any React nodes.
 	unmountComponentAtNode( target );
 	const reboot = reinitializeEditor.bind( null, target, settings );
-	render(
-		<Editor initialSettings={ settings } onError={ reboot } />,
-		target
-	);
+
+	// We dispatch actions and update the store synchronously before rendering
+	// so that we won't trigger unnecessary re-renders with useEffect.
+	{
```

Comment:
I know this has been merged already, but I'm quite curious - why is the code below wrapped in this block?

Reply:
It's just for readability sake, trying to wrap similar things into a block for clarity.